deployment/ccip/changeset/cs_update_rmn_config.go

```diff
@@ -0,0 +1,422 @@
+		if _, exists := peerIds[node.PeerId]; exists {
+			return fmt.Errorf("peerId %x is duplicated", node.PeerId)
+		}
+		peerIds[node.PeerId] = struct{}{}
+
+		if _, exists := offchainPublicKeys[node.OffchainPublicKey]; exists {
+			return fmt.Errorf("offchainPublicKey %x is duplicated", node.OffchainPublicKey)
+		}
+		offchainPublicKeys[node.OffchainPublicKey] = struct{}{}
+	}
+	rmnHome := state.Chains[c.HomeChainSelector].RMNHome
+
+	if rmnHome == nil {
+		return fmt.Errorf("RMNHome not found for chain %d", c.HomeChainSelector)
+	}
+
+	currentDigest, err := rmnHome.GetCandidateDigest(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get RMNHome candidate digest: %w", err)
+	}
+
+	if currentDigest != c.DigestToOverride {
+		return fmt.Errorf("current digest (%x) does not match digest to override (%x)", currentDigest[:], c.DigestToOverride[:])
+	}
+
+	return nil
+}
+
+type PromoteRMNHomeCandidateConfig struct {
+	HomeChainSelector uint64
+	DigestToPromote   [32]byte
+	MCMSConfig        *MCMSConfig
+}
+
+func (c PromoteRMNHomeCandidateConfig) Validate(state CCIPOnChainState) error {
+	err := deployment.IsValidChainSelector(c.HomeChainSelector)
+	if err != nil {
+		return err
+	}
+
+	rmnHome := state.Chains[c.HomeChainSelector].RMNHome
+	if rmnHome == nil {
+		return fmt.Errorf("RMNHome not found for chain %d", c.HomeChainSelector)
+	}
+
+	currentCandidateDigest, err := rmnHome.GetCandidateDigest(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get RMNHome candidate digest: %w", err)
+	}
+
+	if currentCandidateDigest != c.DigestToPromote {
+		return fmt.Errorf("current digest (%x) does not match digest to promote (%x)", currentCandidateDigest[:], c.DigestToPromote[:])
+	}
+
+	return nil
+}
+
+func NewSetRMNHomeCandidateConfigChangeset(e deployment.Environment, config SetRMNHomeCandidateConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+
+	err = config.Validate(state)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	homeChain := e.Chains[config.HomeChainSelector]
+
+	rmnHome := state.Chains[config.HomeChainSelector].RMNHome
+	if rmnHome == nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("RMNHome not found for chain %s", homeChain.String())
+	}
+
+	deployer := getDeployer(e, config.HomeChainSelector, config.MCMSConfig)
+	setCandidateTx, err := rmnHome.SetCandidate(deployer, config.RMNStaticConfig, config.RMNDynamicConfig, config.DigestToOverride)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("build RMNHome set candidate calldata for chain %s: %w", homeChain.String(), err)
+	}
+
+	if config.MCMSConfig == nil {
+		chain := e.Chains[config.HomeChainSelector]
+		_, err := chain.Confirm(setCandidateTx)
+
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to confirm tx for chain %s: %w", homeChain.String(), err)
```

Comment:
For confirm tx let's use `deployment.maybeDataError` it will show you error reason if there is any
```suggestion
			return deployment.ChangesetOutput{}, fmt.Errorf("failed to confirm tx for chain %s: %w", homeChain.String(), deployment.maybeDataError(err))
```

Reply:
Can we include some logs here to denote it's successful